### PR TITLE
Introduce linter and fix all linting issues

### PR DESCRIPTION
**Pull request by jeremyharisch**
Introduced make lint, and fixed all linting issues in the repo

### CODE REVIEW

File: kcp/Makefile [hunk 1]
```diff
@@ -135,4 +135,4 @@ fmt: ## Run go fmt against code.
 .PHONY: lint
 lint: ## Run golangci-lint against code.
 	GOBIN=$(LOCALBIN) go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(GOLANG_CI_LINT_VERSION)
-	$(LOCALBIN)/golangci-lint run
\ No newline at end of file
+	$(LOCALBIN)/golangci-lint run --max-issues-per-linter=0 --max-same-issues=0 --sort-results --fix
```
Comment (adityabhatia) [inline kcp/Makefile:138]: all flags can go to the config file. Please check kyma-operator for reference

File: kcp/main.go [hunk 2]
```diff
@@ -41,14 +41,18 @@ var (
 	setupLog = ctrl.Log.WithName("setup")
 )
 
-func init() {
+func initializeOperator() {
```
Comment (adityabhatia) [inline kcp/main.go:44]:
```suggestion
//nolint:gochecknoinits
func init() {
```

File: kcp/main.go [hunk 3]
```diff
@@ -41,14 +41,18 @@ var (
...
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
 	utilruntime.Must(componentv1alpha1.AddToScheme(scheme))
 	//+kubebuilder:scaffold:scheme
+
+	//+kubebuilder:scaffold:scheme
```
Comment (adityabhatia) [inline kcp/main.go:50]: duplicate

File: skr/pkg/factory/factory.go [hunk 4]
```diff
@@ -10,13 +13,12 @@ import (
...
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/source"
-	"strings"
-	"time"
 )
 
 func InformerFactoryWithLabel(client dynamic.Interface, mgr ctrl.Manager, label, value string) (dynamicinformer.DynamicSharedInformerFactory, error) {
 	// Create informerFactory for each configured label
-	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(client, time.Minute*30, "", func(options *metav1.ListOptions) {
+	defaultResync := time.Minute * 30
```
Comment (adityabhatia) [inline skr/pkg/factory/factory.go:20]: not sure how lint issues passed here. Didn't it report of a magic number error?

**Review by adityabhatia — CHANGES_REQUESTED** [review]
looks great overall, with some minor changes

File: skr/pkg/factory/factory.go (hunk 4 above)
Comment (jeremyharisch) [reply skr/pkg/factory/factory.go:20]: If you just initialise it before using it, then the error disappears

**Review by adityabhatia — APPROVED** [review]
🔥

**Review by adityabhatia — APPROVED** [review]

**Pull request merged**